controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -534,6 +534,13 @@ func (r *BareMetalHostReconciler) actionDeleting(prov provisioner.Provisioner, i
 		return actionError{errors.Wrap(err, "failed to remove finalizer")}
 	}
 
+	secretManager := secretutils.NewSecretManager(info.log, r.Client, r.APIReader)
+
+	err = secretManager.ReleaseSecret(info.bmcCredsSecret)
```

Comment:
This must be done before the finalizer is removed from the BMH, otherwise the finalizer on the Secret could be left behind forever.

Reply:
fixed